src/components/ItemsList.js

```diff
@@ -23,13 +32,33 @@ const ItemsList = () => {
     { idField: 'documentId' },
   );
 
-
   const markItemAsPurchased = (index) => {
     const { items, documentId } = shoppingList[0];
-
-    items[index].lastPurchasedOn = items[index].lastPurchasedOn
-      ? null
-      : Date.now();
+    const { lastPurchasedOn } = items[index];
+    const shoppingItemObject = items[index];
+    if (lastPurchasedOn === null) {
+      shoppingItemObject.lastPurchasedOn = Date.now();
+      shoppingItemObject.numberOfPurchases++;
+      shoppingItemObject.daysLeftForNextPurchase = calculateEstimate(
+        undefined,
+        shoppingItemObject.daysLeftForNextPurchase,
+        shoppingItemObject.numberOfPurchases,
+      );
+    } else {
+      if (wasItemPurchasedWithinLastOneDay(lastPurchasedOn)) {
+        return;
+      } else {
+        shoppingItemObject.numberOfPurchases++;
+        const { daysLeftForNextPurchase } = shoppingItemObject;
+        const now = Date.now();
+        shoppingItemObject.daysLeftForNextPurchase = calculateEstimate(
+          daysLeftForNextPurchase,
+          getDaysBetweenCurrentAndPreviousPurchase(lastPurchasedOn, now),
+          shoppingItemObject.numberOfPurchases,
+        );
+        shoppingItemObject.lastPurchasedOn = now;
```

Comment:
This is awesome. `calculateEstimate` function has all three @params: 
- last estimate purchase interval (in our case named as: `daysLeftForNextPurchase`)
- The number of days between the most recent and previous purchases (in our case named as: `getDaysBetweenCurrentAndPreviousPurchase `)
- Total number of purchases for the item (in our case named as: `numberOfPurchases`)

Great implementation! 🙌

Reply:
Thanks @Lukabaramishvili for the review. Essentially for the first purchase, calling `calculateEstimate` is not necessary at all because the function  returns the same value passed if `numberOfPurchases < 2`. I am surprised you caught it. We shall remove it in our final follow up to the review comments.

Reply:
@nibble0101 Sounds great. Thank you, Joseph.